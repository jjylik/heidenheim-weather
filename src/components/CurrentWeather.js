import React from "react";
import { useSpring, animated } from "react-spring";
import { API_ICON_TO_WEATHER_ICON } from "../constants/icons";
import getCardinal from "../utils/cardinal";
import WeatherIcon from './WeatherIcon'
import WeatherConditionItem from './WeatherConditionItem'

const CurrentWeather = ({ currentWeather }) => {
  const opacityAnimation = useSpring({
    opacity: 1,
    delay: 100,
    from: { opacity: 0 }
  });
  const summaryIcon =
    API_ICON_TO_WEATHER_ICON[currentWeather.icon] || "wi-day-sunny";
  return (
    <>
      <animated.h1>Commuting weather in Heidenheim</animated.h1>
      <WeatherIcon iconName={summaryIcon} />
      <animated.h2 style={opacityAnimation}>
        {currentWeather.summary}
      </animated.h2>
      <div className="weather-item-container">
        <WeatherConditionItem
          label="Temperature"
          value={currentWeather.temperature}
          icon="wi-thermometer"
          round={val => Math.round(val)}
        >
          <span>Â°C</span>
        </WeatherConditionItem>
        <WeatherConditionItem
          label="Chance of rain"
          value={currentWeather.precipProbability * 100}
          icon="wi-rain"
          delay={2000}
          round={val => Math.floor(val)}
        >
          <span>%</span>
        </WeatherConditionItem>
        <WeatherConditionItem
          label="Wind speed"
          value={currentWeather.windSpeed}
          delay={3000}
          icon="wi-strong-wind"
          round={val => Math.round(val * 100) / 100}
        >
          <span>m/s from </span>
          <span>{getCardinal(currentWeather.windBearing)}</span>
        </WeatherConditionItem>
        <WeatherConditionItem
          label="UV index"
          delay={4000}
          value={currentWeather.uvIndex}
          icon="wi-barometer"
          round={val => Math.floor(val)}
        ></WeatherConditionItem>
      </div>
    </>
  );
};

export default CurrentWeather;
