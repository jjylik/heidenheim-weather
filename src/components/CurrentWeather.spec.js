import React from 'react';
import CurrentWeather from './CurrentWeather';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const currentWeather = {
    uvIndex: 0,
    temperature: 12.3,
    summary: "Partially cloudy",
    icon: "cloudy",
    precipProbability: 0.3
  }
  const tree = renderer
    .create(<CurrentWeather currentWeather={currentWeather} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});