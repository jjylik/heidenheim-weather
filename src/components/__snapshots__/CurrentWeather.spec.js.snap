// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h1>
    Commuting weather in Heidenheim
  </h1>,
  <i
    className="wi wi-day-cloudy"
  />,
  <h2
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    Partially cloudy
  </h2>,
  <div
    className="weather-item-container"
  >
    <div
      className="weather-row"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <i
        className="wi wi-thermometer"
      />
      <span>
        Temperature
      </span>
      <div>
        <span>
          0
        </span>
        <span>
          Â°C
        </span>
      </div>
    </div>
    <div
      className="weather-row"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <i
        className="wi wi-rain"
      />
      <span>
        Chance of rain
      </span>
      <div>
        <span>
          0
        </span>
        <span>
          %
        </span>
      </div>
    </div>
    <div
      className="weather-row"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <i
        className="wi wi-strong-wind"
      />
      <span>
        Wind speed
      </span>
      <div>
        <span>
          0
        </span>
        <span>
          m/s from 
        </span>
        <span>
          NW
        </span>
      </div>
    </div>
    <div
      className="weather-row"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <i
        className="wi wi-barometer"
      />
      <span>
        UV index
      </span>
      <div>
        <span>
          0
        </span>
      </div>
    </div>
  </div>,
]
`;
